substrate/frame/staking/src/lib.rs

```diff
@@ -845,6 +851,7 @@ pub trait EraPayout<Balance> {
 	fn era_payout(
 		total_staked: Balance,
 		total_issuance: Balance,
+		max_staking_payout: Percent,
```

Comment:
I invite you to document this trait to say what each parameter is doing. We might then conclude that we are polluting a _trait's API_ based on _one_ of the implementations. 

Nonetheless, given that we have agreed to rewrite these pallets it is fine.

Reply:
In fact, you wouldn't have needed this at all if you had used `parameter_types! {storgae}` to store the percentage, right? https://github.com/paritytech/polkadot-sdk/pull/1660/files#r1335491918

Reply:
> In fact, you wouldn't have needed this at all if you had used `parameter_types! {storgae}` to store the percentage, right? https://github.com/paritytech/polkadot-sdk/pull/1660/files#r1335491918

That's correct. One of the main goals though is to expose the min % of rewards going directly to the treasury through governance, and that value needs to be propagated from the staking pallet to the `era_payout` impl. So I think that the trait needs to be changed, unless I'm missing something?

We could also keep the current `EraPayout` trait and formula as-is and then adjust the payout in the caller site (i.e. the staking pallet), based on the `MaxStakersPayout`. That's basically what we had in the 1st version of the PR but it also doesn't feel super clean.

Reply:
I changed the PR so that the `EraPayout` trait does not need to change. The adjustments to the inflation destination (i.e. validators rewards or remainder aka treasury) are now performed at the staking pallet level, not the era_payout implementation.

This way we don't pollute the EraPayout trait with staking-related logic.